.github/workflows/tests.yml

```diff
@@ -17,14 +17,37 @@ jobs:
 
     steps:
     - uses: actions/checkout@v4
+
     - name: Set up Python ${{ matrix.python-version }}
       uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python-version }}
         cache: "pip"
         allow-prereleases: true
 
-    - name: Run tests - Python ${{ matrix.python-version }}
+    - name: Install dependencies
       run: |
         pip install .[test]
+
+    - name: Run flake8
```

Comment:
I think it's way more coinvent to just add flake8 to the pre-commit configuration.

```yaml
-   repo: https://github.com/PyCQA/flake8
    rev: 7.1.1
    hooks:
    -   id: flake8
```
See https://flake8.pycqa.org/en/latest/user/using-hooks.html#usage-with-the-pre-commit-git-hooks-framework

Advantage of pre-commit it can also be run locally.

Reply:
Until someone runs `git commit --no-verify` 🙈 .  We can certainly run it in both places 😄

Reply:
Haven't ever encountered that, but pre-commit.ci should take care of that anyway.

Reply:
Just want to say I have designed a _lot_ of GitHub Actions workflows and every single one of them needs to be maintained. If there are other tools available, I generally recommend those.